.circleci/config.yml

```diff
@@ -0,0 +1,196 @@
+version: 2.1
+parameters:
+  java-tag:
+    type: string
+    default: "17.0.3"
+orbs:
+  build-tools: circleci/build-tools@2.7.0
+executors:
+  unit_tests_executor:
+    docker: # run the steps with Docker
+      - image: cimg/openjdk:<< pipeline.parameters.java-tag >>
+
+  machine_integration_test_exec:
+    machine: # run the steps with Ubuntu VM
+      image: ubuntu-2204:2022.10.1
+    environment:
+      PGHOST: 127.0.0.1
+    resource_class: medium
+
+common_filters: &common_filters
+  filters:
+    tags:
+      only: /.*/
+    branches:
+      ignore:
+        - gh-pages
+
+workflows:
+  version: 2
+  everything:
+    jobs:
+      - build:
+          <<: *common_filters
+      - metrics-aggregator-tests:
+          <<: *common_filters
+          requires:
+            - build
+
+jobs:
+  metrics-aggregator-tests:
+    executor: machine_integration_test_exec
+    parallelism: 1
+    environment:
+      DIRECTORY: metricsaggregator
+      MAVEN_GOAL: verify
+    steps:
+      - setup_and_run_integration_tests
+
+  build:
+    docker: # run the steps with Docker
+      - image: cimg/openjdk:<< pipeline.parameters.java-tag >>
+        environment:
+          # Java can read cgroup. Sadly the cgroup in
+          # CircleCI is wrong. Have to manually set. Nothing to do with surefire
+          # plugin, it has its own JVM. The two of these must add up to a bit less than 4GB.
+          JAVA_TOOL_OPTIONS: -Xmx512m
+    steps: # a collection of executable commands
+      - checkout # check out source code to working directory
+      - install-git-secrets
+      - run:
+          name: build
+          command: cd metricsaggregator && mvn -B clean install -DskipTests
+
+        # Running scan must occur after build
+      - run_git_secrets_scan
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+
+commands:
+  setup_machine:
+    steps:
+      - run: # Useful for verifying default versions on machine image
+          name: Java/Maven/Python versions
+          command: |
+            java -version
+            mvn -v
+            python3 -V
+      - run:
+          name: Docker-Compose
+          command: cd $DIRECTORY && docker-compose up -d
```

Comment:
All of this is extremely impressive and thorough! If I had written this, I probably would been lazy and not actually run the webservice as part of my testing. :)